flask_resty/jwt.py

```diff
@@ -142,16 +142,18 @@ def decode_token(self, token):
         unverified_header = jwt.get_unverified_header(token)
         jwk = self.get_jwk_for_token(token)
 
-        # It's safe to use alg from the header here,
-        # as we verify that against the algorithm whitelist
+        # It's safe to use alg from the header here, as we verify that against
+        # the algorithm whitelist.
         alg = jwk['alg'] if 'alg' in jwk else unverified_header['alg']
 
-        # jwt.decode will also check this, so this is more defensive.
+        # jwt.decode will also check this, but this is more defensive.
         if alg not in args['algorithms']:
             raise InvalidAlgorithmError(
-                'The specified alg value is not allowed',
+                "The specified alg value is not allowed",
             )
 
-        args['key'] = self.key_from_jwk(jwk, self.algorithms[alg])
-
-        return jwt.decode(token, **args)
+        return jwt.decode(
+            token,
+            key=self.get_key_from_jwk(jwk, self.algorithms[alg]),
+            **args
```

Comment:
does this get overridden if you provided a key in the config somewhere?

Reply:
```
In [1]: def foo(**kwargs): pass

In [2]: foo(key=1, **{'key': 2})
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-74b1f61e9cb5> in <module>()
----> 1 foo(key=1, **{'key': 2})

TypeError: foo() got multiple values for keyword argument 'key'
```

And this will correspondingly be a 500, which I think is what we want.

Reply:
nice. 👍